File: server/router/gb28181/region.go
package gb28181

import (
	"wvp-go/server/api/v1/gb28181"

	"github.com/gin-gonic/gin"
)

type RegionRouter struct{}

var regionRouter = new(RegionRouter)

func (r *RegionRouter) InitRegionRouter(Router *gin.RouterGroup) {
	rg := Router.Group("/region")
	{
		rg.GET("/tree", gb28181.GetRegionTree)
		rg.GET("/list", gb28181.GetRegionList)
		rg.POST("", gb28181.CreateRegion)
		rg.PUT("/:code", gb28181.UpdateRegion)
		rg.DELETE("/:code", gb28181.DeleteRegion)
	}
}

File: web/src/api/region.js
import service from '@/utils/request'

// 获取区域树
export const getRegionTree = () => {
  return service({
    url: '/region/tree',
    method: 'get'
  })
}

// 获取区域列表
export const getRegionList = (params) => {
  return service({
    url: '/region/list',
    method: 'get',
    params
  })
}

// 创建区域
export const createRegion = (data) => {
  return service({
    url: '/region',
    method: 'post',
    data
  })
}

// 更新区域
export const updateRegion = (code, data) => {
  return service({
    url: `/region/${code}`,
    method: 'put',
    data
  })
}

// 删除区域
export const deleteRegion = (code) => {
  return service({
    url: `/region/${code}`,
    method: 'delete'
  })
}
